internal/dataplane/kong_client_test.go

```diff
@@ -357,19 +356,150 @@ func TestKongClientUpdate_WhenNoChangeInConfigNoClientGetsCalled(t *testing.T) {
 	updateStrategyResolver.assertNoUpdateCalled()
 }
 
+type mockConfigStatusQueue struct {
+	wasNotified bool
+}
+
+func newMockConfigStatusQueue() *mockConfigStatusQueue {
+	return &mockConfigStatusQueue{}
+}
+
+func (m *mockConfigStatusQueue) NotifyConfigStatus(context.Context, clients.ConfigStatus) {
+	m.wasNotified = true
+}
+
+func (m *mockConfigStatusQueue) WasNotified() bool {
+	return m.wasNotified
+}
+
+type mockKongConfigBuilder struct {
+	translationFailuresToReturn []failures.ResourceFailure
+}
+
+func newMockKongConfigBuilder() *mockKongConfigBuilder {
+	return &mockKongConfigBuilder{}
+}
+
+func (p *mockKongConfigBuilder) BuildKongConfig() parser.KongConfigBuildingResult {
+	return parser.KongConfigBuildingResult{
+		KongState:           &kongstate.KongState{},
+		TranslationFailures: p.translationFailuresToReturn,
+	}
+}
+
+func (p *mockKongConfigBuilder) returnTranslationFailures(enabled bool) {
+	if enabled {
+		// Return some mocked translation failures.
+		p.translationFailuresToReturn = []failures.ResourceFailure{
+			lo.Must(failures.NewResourceFailure("some reason", &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "name",
+					Namespace: "namespace",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+			},
+			)),
+		}
+	} else {
+		p.translationFailuresToReturn = nil
+	}
+}
+
+func TestKongClientUpdate_ConfigStatusIsAlwaysNotified(t *testing.T) {
+	t.Parallel()
```

Comment:
I just run this test and its runtime couldn't be registered by `go test` (rounding error), grepped to not account for launching test overhead. Do we need the `t.Parallel()` in this case?

```
go test -count 1 -v ./internal/dataplane | grep TestKongClientUpdate_ConfigStatusIsAlwaysNotified
=== RUN   TestKongClientUpdate_ConfigStatusIsAlwaysNotified
=== RUN   TestKongClientUpdate_ConfigStatusIsAlwaysNotified/success
=== RUN   TestKongClientUpdate_ConfigStatusIsAlwaysNotified/gateway_failure
=== RUN   TestKongClientUpdate_ConfigStatusIsAlwaysNotified/translation_failures
=== RUN   TestKongClientUpdate_ConfigStatusIsAlwaysNotified/konnect_failure
=== RUN   TestKongClientUpdate_ConfigStatusIsAlwaysNotified/both_gateway_and_konnect_failure
=== RUN   TestKongClientUpdate_ConfigStatusIsAlwaysNotified/translation_failures_and_konnect_failure
--- PASS: TestKongClientUpdate_ConfigStatusIsAlwaysNotified (0.00s)
    --- PASS: TestKongClientUpdate_ConfigStatusIsAlwaysNotified/success (0.00s)
    --- PASS: TestKongClientUpdate_ConfigStatusIsAlwaysNotified/gateway_failure (0.00s)
    --- PASS: TestKongClientUpdate_ConfigStatusIsAlwaysNotified/translation_failures (0.00s)
    --- PASS: TestKongClientUpdate_ConfigStatusIsAlwaysNotified/konnect_failure (0.00s)
    --- PASS: TestKongClientUpdate_ConfigStatusIsAlwaysNotified/both_gateway_and_konnect_failure (0.00s)
    --- PASS: TestKongClientUpdate_ConfigStatusIsAlwaysNotified/translation_failures_and_konnect_failure (0.00s)
```